src/popper/runner_host.py

```diff
@@ -229,6 +228,155 @@ def _find_container(self, cid):
+            if ecode != 0:
+                log.warning(f"Failed to stop the {c} container")
+
+    def _find_container(self, cid):
+        """Checks whether the container exists."""
+        cmd = ["podman", "inspect", "-f", "{{.Id}}", cid]
+        _, ecode, containers = HostRunner._exec_cmd(cmd, logging=False)
+
+        if ecode == 125:
+            return None
+
+        if ecode != 0:
+            log.fail(f"podman inspect fail: {containers}")
+
+        return containers.strip()
+
+    def _create_container(self, cid, step):
+        build, _, img, tag, build_ctx_path = self._get_build_info(step)
+
+        if build:
+            log.info(f"[{step.id}] podman build {img}:{tag} {build_ctx_path}")
+            if not self._config.dry_run:
+                pass
+        elif not self._config.skip_pull and not step.skip_pull:
+            log.info(f"[{step.id}] podman build {img}:{tag}")
+            if not self._config.dry_run:
+                cmd = ["podman", "pull", f"{img}:{tag}"]
+                HostRunner._exec_cmd(cmd, logging=False)
+
+        if self._config.dry_run:
+            return
+
+        container_args = self._get_container_kwargs(step, f"{img}:{tag}", cid)
+        log.debug(f"Container args: {container_args}")
+
+        msg = [f"{step.id}", "podman", "create", f"name={cid}"]
+        msg.append(f"image={container_args.get('image')}")
+        msg.append(f"entrypoint={container_args.get('entrypoint')}" or "")
+        msg.append(f"command={container_args.get('command')}" or "")
+        log.info(msg)
+
+        cmd = ["podman", "create"]
+
+        cmd.extend(["--name", container_args.get("name") or ""])
+        cmd.extend(["-v", container_args.get("volumes")[0] or ""])
+
+        env = container_args.get("environment")
+        if env:
+            for i, j in env.items():
+                cmd.extend(["-e", f"{i}={j}"])
+
+        cmd.extend(["-d" if container_args.get("detach") else ""])
+
+        cmd.extend(["-w", container_args.get("working_dir")])
+
+        h = container_args.get("hostname", None)
+        if h:
+            cmd.extend(["-h", h])
+
+        domain_name = container_args.get("domainname", None)
+        if domain_name:
+            cmd.extend(["--domainname", domain_name])
+
+        tty = container_args.get("tty", None)
+        if tty:
+            cmd.extend(["-t", tty])
+
+        entrypoint = container_args.get("entrypoint")
+        if entrypoint:
+            cmd.extend(["--entrypoint", entrypoint[0]])
+            entrypoint_rmd = entrypoint[1:]
+
+        cmd.append(container_args.get("image"))
+
+        if entrypoint:
+            for i in entrypoint_rmd:
+                cmd.extend([i or ""])
+
+        for i in container_args["command"]:
+            cmd.extend([i or ""])
+
+        _, ecode, container = HostRunner._exec_cmd(cmd, logging=False)
+
+        if ecode != 0:
+            return None
+
+        container = container.rsplit()
+        return container[-1]
```

Comment:
this might fail if `rsplit()` returns an empty list. To avoid that scenario, the code can check the size of the list and add logic for handling the case of an empty list